1-ECommerce/src/components/Products.tsx

```diff
@@ -0,0 +1,91 @@
+import { useNavigate, useSearchParams } from "react-router-dom";
+
+import { Button } from "./Button";
+import { NothingToShow } from "./NothingToShow";
+import { ProductCard } from "./ProductCard";
+import type { Product } from "./ProductsContainer";
+
+interface ProductProps {
+  products: Product[];
+  favorites: Product[];
+  onFavoriteClick: (product: Product) => void;
+}
+
+type SortProductFunction = (a: Product, b: Product) => number;
+
+const sortAZ = (a: Product, b: Product): number =>
+  a.title.localeCompare(b.title, "en", { sensitivity: "base" });
+
+const sortZA = (a: Product, b: Product): number =>
+  b.title.localeCompare(a.title, "en", { sensitivity: "base" });
+
+const sortLowPrice = (a: Product, b: Product): number => b.price - a.price;
+const sortHighPrice = (a: Product, b: Product): number => a.price - b.price;
+
+const sortingFunctions = {
+  az: sortAZ,
+  za: sortZA,
+  lowPrices: sortLowPrice,
+  highPrices: sortHighPrice,
+};
+
+export const Products = ({
+  products,
+  favorites,
+  onFavoriteClick,
+}: ProductProps) => {
+  const [queryParams, setQueryParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const API_TOTAL_PRODUCTS = 100;
+  const PRODUCTS_PER_PAGE = 6;
+  const currentLimit = Number(queryParams.get("limit") ?? 6);
+
+  const view = String(queryParams.get("view"));
+
+  const sortedProducts =
+    view in sortingFunctions
+      ? products.toSorted(sortingFunctions[view] as SortProductFunction)
+      : products;
```

Comment:
I think this could introduce a "bug" later on. Even though I'm not mutating `products` (which should be read-only) or `sortedProducts` in other parts of the code, `sortedProducts` sometimes holds a reference to `products` and sometimes holds a sorted copy. So, in the future, it won't be "safe" to mutate `sortedProducts`, because there's no certainty if it's a different object from the `products` prop. Instead of using `: products` perhaps it would be best to use `: [...products]`